target_chains/ethereum/contracts/contracts/pyth/PythGovernance.sol

```diff
@@ -190,4 +201,44 @@ abstract contract PythGovernance is
 
         emit ValidPeriodSet(oldValidPeriod, validTimePeriodSeconds());
     }
+
+    function setWormholeAddress(
+        SetWormholeAddressPayload memory payload,
+        bytes memory encodedVM
+    ) internal {
+        address oldWormholeAddress = address(wormhole());
+        setWormhole(payload.newWormholeAddress);
```

Comment:
Would it be safer to do 
```
(IWormhole.VM memory vm, bool valid, ) = IWormhole(payload.newWormholeAddress).parseAndVerifyVM(
            encodedVM
        );
// validation checks
setWormhole(payload.newWormholeAddress);
emit WormholeAddressSet(oldWormholeAddress, address(wormhole()));
```

Reply:
I think these are equally safe. Checking using the `wormhole()` after setting it just makes sure we use the same command we use in other places for the sanity check.